AccountingSystem/Agent/NetworkAgent.py

```diff
@@ -160,62 +160,79 @@ def processMessage( self, headers, body ):
 
-    if body.has_key( 'summaries' ):
-      for summary in body['summaries']:
-
-        # we are interested in 5 minutes summaries only
-        if summary['summary_window'] == '300':
-          for data in summary['summary_data']:
-            try:
-
-              # set the time and create a hash
-              startTime = datetime.utcfromtimestamp( float( data[0] ) )
-              endTime = datetime.utcfromtimestamp( float( data[0] ) + 300 )
-
-              key = metadataKey
-              key += str( startTime )
-              key += str( endTime )
-
-              # use existing or create a new temporary accounting object to store the data in DB
-              if self.buffer.has_key( key ):
-                net = self.buffer[key]['object']
-              else:
-                net = Network()
-                net.setStartTime( startTime )
-                net.setEndTime( endTime )
-                net.setValuesFromDict( metadata )
-
-              # look for supported event types
-              if summary['event_type'] == 'packet-loss-rate':
-                self.PLRMetricCount += 1
-                net.setValueByKey( 'PacketLossRate', data[1] * 100 )
-              elif summary['event_type'] == 'histogram-owdelay' and summary['summary_type'] == 'statistics':
-                self.OWDMetricCount += 1
-
-                # skip metrics with invalid data
-                if data[1]['mean'] < 0 or data[1]['minimum'] < 0 or data[1]['maximum'] < 0:
-                  raise Exception( 'Invalid OWD metric (%s, %s, %s)' % ( data[1]['mean'], data[1]['minimum'], data[1]['maximum'] ) )
-                else:
-                  # approximate jitter value as OWDMax - OWDMin
-                  net.setValueByKey( 'Jitter', data[1]['maximum'] - data[1]['minimum'] )
-                  net.setValueByKey( 'OneWayDelay', data[1]['mean'] )
-              else:
-                self.skippedMetricCount += 1
-                continue
-              
-              self.buffer[key] = {'addTime': datetime.now(), 'object': net}
-
-            # suppress all exceptions to protect the listener thread
-            except Exception as e:
-              self.skippedMetricCount += 1
-              self.log.warn( 'Metric skipped because of an exception: %s' % e )
+    timestamps = sorted( body['datapoints'] )
+    
+    # approximate the value of first the timestamp
+    # (datapoints are take every 1 minute)
+    previousTimestamp = float( timestamps[0] ) - 60
+    for nextTimestamp in timestamps:
+    
+      data = body['datapoints'][nextTimestamp]
+
+      try:
+        # set the time and create a hash
+        startTime = datetime.utcfromtimestamp( previousTimestamp )
+        endTime = datetime.utcfromtimestamp( float( nextTimestamp ) )
 
+        key = "%s%s%s" % ( metadataKey, str( startTime ), str( endTime ) )
+
+        # use existing or create a new temporary accounting
+        # object to store the data in DB
+        if self.buffer.has_key( key ):
```

Comment:
```python
if key in self.buffer:
```
Could you please also use meaning full names instead of key?

Reply:
Sure - I have just updated this and some other things.